content/rooms/fragments/v1-stringy-power-levels.md

```diff
@@ -0,0 +1,24 @@
+---
+toc_hide: true
+---
+
+##### Power levels accepted as strings
+
+In order to maintain backwards compatibility with early implementations,
+power levels can optionally be represented in string format instead of
+integer format. A homeserver must be prepared to deal with this by parsing
+the power level from a string. In these cases, the following formatting of the
+power level string is allowed:
```

Comment:
formatting:

```suggestion
In order to maintain backwards compatibility with early implementations,
each of the integer-valued properties within 
[`m.room.power_levels`](/client-server-api#mroompower_levels) events can
be encoded as strings instead of integers. This includes the nested values
within the `events`, `notifications` and `users` properties.

For example, the following is a valid `m.room.power_levels` event in this room version:


```json
{
  "content": {
    "ban": "50",
    "events": {
      "m.room.power_levels": "100"
    },
    "events_default": "0",
    "state_default": "50",
    "users": {
      "@example:localhost": "100"
    },
    "users_default": "0"
  },
  "origin_server_ts": 1432735824653,
  "room_id": "!jEsUZKDJdhlrceRyVU:example.org",
  "sender": "@example:example.org",
  "state_key": "",
  "type": "m.room.power_levels"
}
```

When the value is an integer, they must be the following format:
```

Reply:
ugh, github.